File: src/main/java/myLang/response/Response.java
package myLang.response;

/**
 * Интерфейс реализует ответ всех обработчиков интерфейса {@link myLang.ContextHandler}.
 * Реализован для унификации всех ответов.
 *
 * @param <T> тип возвращаемого в ответе значения
 * @author Ivan Kylchik
 */
public interface Response<T> {
    T getResponse();
}

File: src/main/kotlin/myLangKotlin/response/Response.kt
package myLangKotlin.response

/**
 * Интерфейс реализует ответ всех методов visitor.
 * Реализован для унификации всех ответов.
 *
 * @param <T> тип возвращаемого в ответе значения
 * @author Ivan Kylchik
 */
interface Response<T> {
    var response: T
}

